hyperparameter_tuning/tensorflow_mnist/hpo_tensorflow_mnist.ipynb

```diff
@@ -179,10 +179,11 @@
     "estimator = TensorFlow(entry_point='mnist.py',\n",
```

Comment:
<p>I can see that you updated the code to adapt sagemaker v2, but the entire file <code>mnist.py</code>  is out of date. As a result, the notebook would not run. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/1865/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Reply:
It works for me in a SageMaker notebook.
I checked the failed build logs and the failure looks like a build infra limit problem, not a notebook issue.

`com.amazonaws.sagemaker.api.ResourceLimitExceeded: The account-level service limit 'total number of notebook instances' is 40 NotebookInstances, with current utilization of 40 NotebookInstances and a request delta of 1 NotebookInstances. Please contact AWS support to request an increase for this limit.`

Reply:
Hey @gilinachum, I got the following exception after the cell `tuner.fit()`
```
UnexpectedStatusException                 Traceback (most recent call last)
<ipython-input-10-a8de378f1c7e> in <module>
----> 1 tuner.fit(inputs)

~/.local/lib/python3.6/site-packages/sagemaker/tuner.py in fit(self, inputs, job_name, include_cls_metadata, estimator_kwargs, wait, **kwargs)
    447 
    448         if wait:
--> 449             self.latest_tuning_job.wait()
    450 
    451     def _fit_with_estimator(self, inputs, job_name, include_cls_metadata, **kwargs):

~/.local/lib/python3.6/site-packages/sagemaker/tuner.py in wait(self)
   1583     def wait(self):
   1584         """Placeholder docstring."""
-> 1585         self.sagemaker_session.wait_for_tuning_job(self.name)
   1586 
   1587 

~/.local/lib/python3.6/site-packages/sagemaker/session.py in wait_for_tuning_job(self, job, poll)
   3182         """
   3183         desc = _wait_until(lambda: _tuning_job_status(self.sagemaker_client, job), poll)
-> 3184         self._check_job_status(job, desc, "HyperParameterTuningJobStatus")
   3185         return desc
   3186 

~/.local/lib/python3.6/site-packages/sagemaker/session.py in _check_job_status(self, job, desc, status_key_name)
   3265                 ),
   3266                 allowed_statuses=["Completed", "Stopped"],
-> 3267                 actual_status=status,
   3268             )
   3269 

UnexpectedStatusException: Error for HyperParameterTuning job sagemaker-tensorflow-201216-1938: Failed. Reason: No objective metrics found after running 5 training jobs. Please ensure that the custom algorithm is emitting the objective metric as defined by the regular expression provided.
```

The error does make sense, because in `mnist.py`, I didn't see any print statement about the model's performance  metrics.

Reply:
My fix gets the job started (before it wasn't). But it seems like the whole notebook needs to be redone.  I'll close the PR. and I hope I can find the time to redo this, and move the TF script to script mode.